### DISCUSSION TITLE
Alterar o framework ou criar um "custom.css" ?? :)

### DISCUSSION THREAD

**Discussion by andre-mendes** [frontendbr/forum#195] [Perguntas e Respostas]
Fala galera, blz? :D Tenho uma dúvida...
Por exemplo:

Se eu baixar o Bootstrap-Sass, aí eu vou ter todos os componentes separados, bonitinhos...

Daí vamos supor que eu queira alterar as propriedades dos botões...

Seria correto eu abrir o buttons.scss do bootstrap e alterar direto lá... Ou eu abrir um outro arquivo, tipo "my-personal-bootstrap-changes.scss" e alterar as propriedades lá nesse arquivo separado?

Porque se eu alterar direto no arquivo do bootstrap, e eu quiser atualizar para a versão 4.0 posteriormente, eu vou perder essas alterações! Mas se eu não alterar direto no arquivo do bootstrap, por que eu tenho esses arquivos todos separados? Só para eu poder dar @import somente nos que eu vou usar? Mas não seria correto eu alterar eles? Se eu não alterar ele, eu não estaria reescrevendo código??

Mas e daí como fica o variaveis.scss? Esse eu posso alterar?

Enfim... Qual seria o recomendado pra se trabalhar com isso? :P

**Comment by tiaguinhor** [comment]
Jamais altere o core. Opte sempre por criar métodos de trabalhar com sua customização independente da library.

**Comment by ricardoruwer** [comment]
@tiaguinhor 
Sim, mas o correto seria eu pegar o arquivo do bootstap (por exemplo o buttons.scss), dar um CTRL-C, abrir um arquivo novo, dar CTRL-V e editar ele com minhas configurações pessoais... E aí no meu HTML eu só chamo esse arquivo...

Ou o correto seria eu chamar no HTML o buttons.scss do bootstrap e também um "custom.css" onde eu altero as propriedades do arquivo do bootstrap?
Daí quando o bootstrap lançar uma versão nova, eu substituo o buttons.scss e só faço alterações no meu "custom.css"?

**Comment by felipefialho** [comment]
Mano, hoje não uso mais o Bootstrap, mas usava essa estratégia que você está usando a uns anos atrás (meados de 2013) e até escrevi um artigo sobre isso chamado [Utilizando o Bootstrap de forma consistente](http://www.felipefialho.com/blog/2014/utilizando-o-bootstrap-de-forma-consistente).

Pode alterar o core do projeto sem dó. 

Isso vale para qualquer plugin inclusive. Sempre adiciono o vendor do `.css` dentro do meu stylus e otimizo lá dentro. Tenho pavor de CSS mal desenvolvido (vários níveis de aninhamento por exemplo) ou de carregar coisas desnecessárias.
